### PR TITLE
support share cores with numa binding

### PR DESCRIPTION
**Pull request by nightmeng**
#### What type of PR is this?
Features

#### What this PR does / why we need it:
Support share cores with NUMA binding

### CODE REVIEW

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go [hunk 1]
```diff
@@ -425,7 +426,31 @@ func (cra *cpuResourceAdvisor) assignToRegions(ci *types.ContainerInfo) ([]regio
...
+		return cra.assignDedicatedContainerToRegions(ci)
+	default:
+		return nil, nil
+	}
+}
+
+func (cra *cpuResourceAdvisor) assignShareContainerToRegions(ci *types.ContainerInfo) ([]region.QoSRegion, error) {
+	numaID := cpuadvisor.FakedNUMAID
+	if !cra.conf.GenericSysAdvisorConfiguration.DisableShareCoresNumaBinding && ci.IsNumaBinding() {
+		if ci.OwnerPoolName == "" {
```
Comment (cheney-lin) [inline pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:442]: why not consider rampup？

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/assembler/provisionassembler/assembler_common.go [hunk 2]
```diff
@@ -91,18 +92,40 @@ func (pa *ProvisionAssemblerCommon) AssembleProvision() (types.InternalCPUCalcul
...
+				reservedForReclaim := pa.getNumasReservedForReclaim(r.GetBindingNumas())
 
+				nonReclaimRequirement := int(controlKnob[types.ControlKnobNonReclaimedCPUSize].Value)
+				available := getNumasAvailableResource(*pa.numaAvailable, r.GetBindingNumas())
+				if nodeEnableReclaim {
+					calculationResult.SetPoolEntry(r.OwnerPoolName(), regionNuma, nonReclaimRequirement)
+					reclaimed := available - nonReclaimRequirement + reservedForReclaim
+					calculationResult.SetPoolEntry(state.PoolNameReclaim, regionNuma, reclaimed)
+				} else {
+					calculationResult.SetPoolEntry(r.OwnerPoolName(), regionNuma, available)
```
Comment (cheney-lin) [inline pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/assembler/provisionassembler/assembler_common.go:106]: what about reservedForReclaim?

**Comment by waynepeking348** [comment]
what's the difference between this pr and https://github.com/kubewharf/katalyst-core/pull/481 ?

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/assembler/provisionassembler/assembler_common.go (hunk 2 above)
Comment (nightmeng) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/assembler/provisionassembler/assembler_common.go]: done

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go (hunk 1 above)
Comment (nightmeng) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:442]: snb doesn't support rampup

File: cmd/katalyst-agent/app/options/sysadvisor/sysadvisor_base.go [hunk 3]
```diff
@@ -31,15 +31,19 @@ import (
 
 // GenericSysAdvisorOptions holds the configurations for sysadvisor
 type GenericSysAdvisorOptions struct {
-	SysAdvisorPlugins  []string
-	StateFileDirectory string
+	SysAdvisorPlugins            []string
+	StateFileDirectory           string
+	ClearStateFileDirectory      bool
+	DisableShareCoresNumaBinding bool
```
[the remark this answers is not in the document]
Comment (nightmeng) [reply cmd/katalyst-agent/app/options/sysadvisor/sysadvisor_base.go:38]: This option is only for rollback

File: pkg/agent/qrm-plugins/cpu/dynamicpolicy/cpueviction/strategy/pressure_load.go [hunk 4]
```diff
@@ -91,19 +91,21 @@ type CPUPressureLoadEviction struct {
...
-		syncPeriod:     conf.LoadEvictionSyncPeriod,
+		state:            state,
+		emitter:          emitter,
+		metaServer:       metaServer,
+		metricsHistory:   make(map[string]Entries),
+		qosConf:          conf.QoSConfiguration,
+		dynamicConf:      conf.DynamicAgentConfiguration,
+		skipPools:        sets.NewString(conf.LoadPressureEvictionSkipPools...),
+		syncPeriod:       conf.LoadEvictionSyncPeriod,
+		configTranslator: general.NewCommonSuffixTranslator("-NUMA"),
```
Comment (waynepeking348) [inline pkg/agent/qrm-plugins/cpu/dynamicpolicy/cpueviction/strategy/pressure_load.go:108]: put `-NUMA` as a configurable params will be better

**Review by cheney-lin — APPROVED** [review]

**Review by waynepeking348 — APPROVED** [review]

**Pull request merged**